### ISSUE TITLE
Debian build: dpkg-genbuildinfo fails because of not having correct package directory

### ISSUE THREAD

**Issue by adriangibanelbtactic** [ONLYOFFICE/document-server-package#233]
**Describe your problem**:

When building Debian package it fails because of dpkg-genbuildinfo not having the correct package directory.

**Steps to reproduce the problem**:
- Build _documentserver_ thanks to _build_tools_ in /root/build_tools directory.

Try to package thanks to:
```
apt install build-essential m4 npm
npm install -g pkg

cd /root
git clone https://github.com/ONLYOFFICE/document-server-package.git
cd document-server-package/deb
apt build-dep ./
cd /root/document-server-package

PRODUCT_VERSION='7.0.0' BUILD_NUMBER='132' make deb
```

**What were you expecting to happen**
The package is built and no error happens.

**What happens instead**

The main package is built but the main build process (Makefile) fails because of genbuildinfo not working properly.

```
dpkg-deb: construyendo el paquete `onlyoffice-documentserver-dbgsym' en `./onlyoffice-documentserver-dbgsym_7.0.0-132
_amd64.deb'.
dpkg-deb: construyendo el paquete `onlyoffice-documentserver' en `./onlyoffice-documentserver_7.0.0-132_amd64.deb'.
make[2]: se sale del directorio '/root/document-server-package/deb'
make[1]: se sale del directorio '/root/document-server-package/deb'
 dpkg-genbuildinfo --build=binary
dpkg-genbuildinfo: fallo: no se puede obtener el estado (fstat) del fichero «../onlyoffice-documentserver-dbgsym_7.0.0-132_amd64.deb»: No such file or directory
dpkg-buildpackage: fallo: dpkg-genbuildinfo --build=binary subprocess returned exit status 255
make: *** [Makefile:445: deb/onlyoffice-documentserver_7.0.0-132_amd64.deb] Error 255
```
**Proposed fix**

At:
https://github.com/ONLYOFFICE/document-server-package/blob/v7.0.0.132/Makefile#L445
you can fix the current line:
`cd deb && dpkg-buildpackage -b -uc -us --changes-option=-u.`
so that it is:
`cd deb && dpkg-buildpackage -b -uc -us --changes-option=-u. --buildinfo-option=-u.`
.

**document-server-package base repo version**: v7.0.0.132

**Operating system**: Debian 11 Netinst

**Compiler version**: N/A.

I'm open to send you a pull request that fixes this problem if you confirm me that the oldest machine where you build Debian packages has `--buildinfo-option` as a dpkg-buildpackage option.

**Comment by ShockwaveNN** [comment]
Hi, I'm personally not very good at building packages, but I'll show this to developers after answering some question

> if you confirm me that the oldest machine where you build Debian packages has

I can confirm that the oldest system we used to build - Ubuntu 14.04. Moreover - all packages we build use this system (we know this is not particularly right, but have not enough resources to change it right now)

I've checked `dpkg-buildpackage` on Ubuntu v14 - it's 
```
Debian dpkg-buildpackage version 1.17.5.
```
And judging by the output of 
```
#dpkg-buildpackage --help | grep buildinfo

```

There is no such option

Any alternatives we can use without loosing Ubuntu 14.04 compatibility?

**Comment by adriangibanelbtactic** [comment]
> I've checked `dpkg-buildpackage` on Ubuntu v14 - it's
> 
> ```
> Debian dpkg-buildpackage version 1.17.5.
> ```
> 
> And judging by the output of
> 
> ```
> #dpkg-buildpackage --help | grep buildinfo
> ```
> 
> There is no such option
> 
> Any alternatives we can use without loosing Ubuntu 14.04 compatibility?

Yes, the alternative is to stop using a custom output directory and use the default `..`.

You currently use `deb/debian` you could use an additional directory and have: `deb/document-server/debian/` instead.
A few lines might need to be modified in Makefile (to copy files into the right place) but this will output your packages into deb directory as you originally did.

**Comment by ShockwaveNN** [comment]
Ok, I've completely out of ideas that this mean )))

I'm calling the build engineer

@agolybev Please take a look at this issue

**Comment by ShockwaveNN** [comment]
We have a talk with @agolybev and described to fill out issue 55086 in our private issue tracker for this problem

We are in process of rewriting and optimizing build process of all our artifacts and hope we can fix your issue in process, but no promises )

Thanks anyway

**Comment by adriangibanelbtactic** [comment]
> We are in process of rewriting and optimizing build process of all our artifacts and hope we can fix your issue in process, but no promises )

I wouldn't have minded to send a pull request with this enhacement.
If you are going to rewrite and optimize the build process it doesn't make sense to even think on doing such a pull request.

Thank you for your feedback.

**Comment by ShockwaveNN** [comment]
Hi this issue should be fixed via https://github.com/ONLYOFFICE/document-server-package/pull/259

Please note that it's only on `develop` branch and this version may include any number of bugs
If everything will be well - this fix will be included in not next, but next-next major version, it may be v7.2

**Comment by ShockwaveNN** [comment]
A little update - those changes will be in v7.1 which I really hope we will release in a month
https://github.com/ONLYOFFICE/document-server-package/pull/265

**Comment by ShockwaveNN** [comment]
The fix is part of v7.1.0 release so issue should be fixed for stable

I'm closing this issue
